src/core/processing/qgsprocessingparameters.cpp

```diff
@@ -3087,14 +3099,21 @@ QgsProcessingOutputDefinition *QgsProcessingParameterFeatureSink::toOutputDefini
 
 QString QgsProcessingParameterFeatureSink::defaultFileExtension() const
 {
-  QgsSettings settings;
-  if ( hasGeometry() )
+  if ( QgsProcessingProvider *p = provider() )
   {
-    return settings.value( QStringLiteral( "Processing/DefaultOutputVectorLayerExt" ), QStringLiteral( "shp" ), QgsSettings::Core ).toString();
+    return p->defaultVectorFileExtension( hasGeometry() );
   }
   else
   {
-    return QStringLiteral( "dbf" );
+    QgsSettings settings;
+    if ( hasGeometry() )
+    {
+      return settings.value( QStringLiteral( "Processing/DefaultOutputVectorLayerExt" ), QStringLiteral( "shp" ), QgsSettings::Core ).toString();
```

Comment:
Maybe it makes sense to check if global default extension supported by provider? Something like
```
QString defaultExtension = settings.value( QStringLiteral( "Processing/DefaultOutputVectorLayerExt" ), QStringLiteral( "shp" ), QgsSettings::Core ).toString();
QStringList supportedExtensions = p->supportedOutputVectorLayerExtensions();
if ( supportedExtensions.contains( defaultExtension ) )
{
  return defaultExtension;
}
else
{
  return supportedExtensions.at( 0 );
}
```
This also applies to raster outputs.
Just thinking aloud.

Reply:
Thanks, done!